### PR TITLE
add additional mutation filtering

### PR DESCRIPTION
**Pull request by bprize15**
resolves https://github.com/oncokb/oncokb/issues/3626

### CODE REVIEW

File: src/main/webapp/app/pages/curation/CurationPage.tsx
Comment (calvinlu3) [inline src/main/webapp/app/pages/curation/CurationPage.tsx:430]: Index shouldn't be `0` here

File: src/main/webapp/app/pages/curation/CurationPage.tsx
Comment (calvinlu3) [inline src/main/webapp/app/pages/curation/CurationPage.tsx:646]: Wonder if we should also have a reset button next to the text `Showing x of y matching search`.

File: src/main/webapp/app/pages/curation/CurationPage.tsx
Comment (calvinlu3) [inline src/main/webapp/app/pages/curation/CurationPage.tsx:657]: Worth to discuss if we want to just have the filter results change as we click on checkboxes or make selections instead of having to make an extra click to apply filters.

File: src/main/webapp/app/pages/curation/CurationPage.tsx
Comment (calvinlu3) [inline src/main/webapp/app/pages/curation/CurationPage.tsx:595]: Can we add the `for` prop and also would need to add and `id` to the Input. I believe it supposed to make it so that when you click on the label, it also checks the box/radio

File: src/main/webapp/app/pages/curation/CurationPage.tsx
Comment (calvinlu3) [inline src/main/webapp/app/pages/curation/CurationPage.tsx:616]: Maybe could shorten to just Therapeutic Levels?

File: src/main/webapp/app/pages/curation/CurationPage.tsx
Comment (calvinlu3) [inline src/main/webapp/app/pages/curation/CurationPage.tsx:627]: I haven't looked at this PR locally, so wondering what the 18.75px is used for? Is it for centering the label?

File: src/main/webapp/app/pages/curation/CurationPage.tsx
Comment (calvinlu3) [inline src/main/webapp/app/pages/curation/CurationPage.tsx:637]: We can probably just use the `btn-outline-primary`, but I see that it also conflicts with the save button. If we just show filtered results immediately, then it would be fine.

**Comment by calvinlu3** [comment]
@bprize15 Just a couple of comments. Would be great to also get Hongxin to review once you've made the changes. Thanks!

File: src/main/webapp/app/pages/curation/CurationPage.tsx
Comment (bprize15) [reply src/main/webapp/app/pages/curation/CurationPage.tsx:627]: The checkbox from reactstrap has a left margin of -18.75px for some reason. This is to account for that and make the spacing even on both sides.

File: src/main/webapp/app/pages/curation/CurationPage.tsx
Comment (bprize15) [reply src/main/webapp/app/pages/curation/CurationPage.tsx:657]: Discussed with Hongxin. We changed it to a modal, so we the confirm button makes the most sense in this case.

**Comment by zhx828** [comment]
- [ ] `Showing ...`  language should only be shown when there is a filter
- [ ] we should differentiate the filter icon when there is/not filter applied (similar to comment, maybe change to yellow when there is a filter)?
- [ ] Add `cursor: pointer;` on the filter icon
- [ ] Are you calculating the stats now? For instance, in ABL1, when I apply LoF, I see no mutation matched. Shouldn't that checkbox be disabled?

File: src/main/webapp/app/pages/curation/CurationPage.tsx
Comment (zhx828) [inline src/main/webapp/app/pages/curation/CurationPage.tsx:68]: This should be a computed value.

File: src/main/webapp/app/pages/curation/CurationPage.tsx
Comment (zhx828) [inline src/main/webapp/app/pages/curation/CurationPage.tsx:76]: This should be computed.

File: src/main/webapp/app/pages/curation/CurationPage.tsx
Comment (zhx828) [inline src/main/webapp/app/pages/curation/CurationPage.tsx:585]: Use class name instead of style?

File: src/main/webapp/app/pages/curation/CurationPage.tsx
Comment (zhx828) [inline src/main/webapp/app/pages/curation/CurationPage.tsx:582]: No need to change, but you can use `0`

File: src/main/webapp/app/pages/curation/CurationPage.tsx
Comment (zhx828) [reply src/main/webapp/app/pages/curation/CurationPage.tsx:612]: Agree with Calvin

**Review by zhx828 — CHANGES_REQUESTED** [review]

File: src/main/webapp/app/pages/curation/CurationPage.tsx
Comment (bprize15) [reply src/main/webapp/app/pages/curation/CurationPage.tsx:585]: I wanted an exact left margin of 18.75 pixels. The checkbox originally comes with a left margin of -18.75px which I needed to set to 0 to fix the spacing. I use the above to offset this. I tried using `ml-3` and `ml-4`, but they both looked off to me. Let me know what you think:
Screenshot 2024-01-02 at 4 54 45 PM
Screenshot 2024-01-02 at 4 54 57 PM

Comment (zhx828) [reply src/main/webapp/app/pages/curation/CurationPage.tsx]: Makes sense. Then could you make it a constant and add a comment?

**Review by zhx828 — APPROVED** [review]
LGTM, thanks! Once the conflict is resolved, feel free to merge.

**Pull request merged**
